.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,59 @@
+name: Run Unit Tests
+
+on: [push, pull_request]
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-20.04
+    services:
+      mariadb:
+        image: mariadb:10.4
+        env:
+          MARIADB_ALLOW_EMPTY_ROOT_PASSWORD: 'yes'
+          MARIADB_DATABASE: apel_unittest
+        ports:
+          - 3306:3306
+        options: >-
+          --health-cmd="mysqladmin ping -h localhost"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=3
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.6', '3.9']
+    name: Python ${{ matrix.python-version }} test
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install system dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y build-essential python3-dev libldap2-dev libsasl2-dev libmysqlclient-dev
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install -r requirements-test.txt
+
+    - name: Pre-test setup
+      run: |
+        export TMPDIR=$PWD/tmp
+        mkdir -p $TMPDIR
+        export PYTHONPATH=$PYTHONPATH:`pwd -P`
+        cd test
+
+    - name: Run unit tests
+      run: coverage run --branch --source=apel,bin -m unittest discover -s test -p 'test_*.py'
```

Comment:
Shouldn't need to add `-s test -p 'test_*.py'`

Reply:
`--buffer` would be a useful option to add.

Reply:
For some reason I couldn't run with the command below. So, I have to specify `-s test` at the least to successfully run the testcases.

Not working in my case:
`coverage run --branch --source=apel,bin -m unittest discover --buffer`


Working if I explicitly specify the -s indicating the start-directory.
`coverage run --branch --source=apel,bin -m unittest discover -s test --buffer`